API.php

```diff
@@ -79,7 +79,17 @@ public function getKeywordContentFromNameId($idSite, $period, $date, $idSubtable
         $campaignName = $row->getColumn('label');
 
         $campaignsDataTable = ReferrersAPI::getInstance()->getCampaigns($idSite, $period, $date, $segment, false);
-        $campaignRow        = $campaignsDataTable->getRowFromLabel($campaignName);
+
+        if ($campaignsDataTable instanceof DataTable\Map) {
+            foreach ($campaignsDataTable->getDataTables() as $table) {
+                $campaignRow = $table->getRowFromLabel($campaignName);
+                if ($campaignRow) {
+                    break;
+                }
+            }
```

Comment:
Is it possible that the same row would exist across different datatables? Eg, if it's a multi period and the campaign name exists for multiple days? Would we need to take this into account?

Reply:
It's only used to determine the correct subtable id if the plugins report is empty and it tries to fall back to the referrers api. Shouldn't the subtable id be the same across tables for different dates? 🤔

Reply:
I think it depends on the period, since it's not generated based on the label. I'm pretty sure it's generated incrementally, so if there's more or different data in one period than another, then the subtable IDs will refer to different labels.

Maybe a test for this that had different data in multiple periods might be a good idea.

Reply:
@diosmosis But wouldn't that mean row evolution can't work for entries in a subtable? 🤔 

Need to postpone adding some possible tests to somewhen else, as the issue actually wasn't planned for the milestone and I'm busy with some other stuff

Reply:
It's been a while since I looked at the row evolution feature, but I think it uses the `label` parameter and `LabelFilter` to get the correct data.

Reply:
Yes it uses the label to filter. But afaik it also uses the subtableId if row evolution is opened for a subtable. Seems that actually doesn't work atm. When opening a row evolution for a subtable if the websites report it currently shows no data at all: https://demo.matomo.cloud/index.php?module=CoreHome&action=index&idSite=1&period=day&date=yesterday#?idSite=1&period=day&date=yesterday&segment=&category=Referrers_Referrers&subcategory=Referrers_SubmenuWebsitesOnly&popover=RowAction$3ARowEvolution$3AReferrers.getUrlsFromWebsiteId$3A$257B$2522idSubtable$2522$253A$25224$2522$257D$3Awww.scubadiving.com$20$3E$20$40tags$252Fscuba-diving-news

Might be the same as https://github.com/matomo-org/matomo/issues/9634

Reply:
I thought it used labels like `supertablelabel > subtablelabel` to get the subtable? In RowEvolution::loadRowEvolutionDataFromAPI eg I see no subtable ID just a label. I could be wrong, but it's been my understanding for a long time now that idSubtables are just incremented in DataTable\Manager not based on the label, so are different based on how they are archived. Which was why our tests remove/ignore idsubtable from api output.